src/object_types/main_menu.lua

```diff
@@ -0,0 +1,328 @@
+        this.options_menu_option = 1
+        this.confirm_menu_option = 1
+        this.confirm_timer = 0
+
+        -- Subscribe to input events.
+        object.subscribe(this, 'inputdown')
+
+        this.load_current_mode = function(self)
+            log.info('Detecting current mode..')
+
+            -- Grab options and state from the current video mode.
+            -- Enumerate available modes on the host.
+            self.modes = {
+                { type = 'windowed' }
+            }
+
+            for k, mode in ipairs(love.window.getFullscreenModes()) do
+                mode.type = 'fullscreen'
+                self.modes[k + 1] = mode
+            end
+
+            -- Locate the current video mode.
+            local w, h, flags = love.window.getMode()
+
+            -- Grab the current flags for vsync and msaa.
+            self.option_vsync = flags.vsync
+            self.option_msaa = flags.msaa
+            self.option_fullscreen = flags.fullscreen
+
+            if self.option_vsync == 0 then
+                self.option_vsync = false
+            end
+
+            log.debug('detected mode %d by %d, fullscreen %s, vsync %s, msaa %d',
+                      w, h, flags.fullscreen, flags.vsync, flags.msaa)
+        end
+
+        -- Helper functions for manipulating modes.
+        this.set_mode = function(_, w, h, flags)
+            love.window.setMode(w, h, flags)
+            love.window.restore()
+
+            w, h, flags = love.window.getMode()
+
+            log.debug('main_menu object set mode: %d by %d, fullscreen %s, vsync %s, msaa %d',
+                      w, h, tostring(flags.fullscreen), tostring(flags.vsync), flags.msaa)
+
+            event.push('fbsize', w, h)
+        end
+
+        -- Helper member function to draw an element with an optional selector.
+        this.draw_element = function(self, str, x, y, font, selected)
+            font = font or self.font
+
+            -- Compute text boundaries.
+            local tw, th = font:getWidth(str), font:getHeight()
+
+            love.graphics.setFont(font)
+
+            -- Render centered option.
+            love.graphics.print(str, x - tw / 2, y - th / 2)
+
+            -- If selected, render selector tris.
+            if selected then
+                love.graphics.polygon('fill', x - tw / 2 - th / 2, y,
+                                              x - tw / 2 - th, y - th / 4,
+                                              x - tw / 2 - th, y + th / 4)
+                love.graphics.polygon('fill', x + tw / 2 + th / 2, y,
+                                              x + tw / 2 + th, y - th / 4,
+                                              x + tw / 2 + th, y + th / 4)
+            end
+        end
+    end,
+
+    inputdown = function(this, key)
+        log.debug('Handling inputdown: %s', key)
+
+        if key == 'crouch' then
+            this.option = this.option + 1
+        end
+
+        if key == 'jump' then
```

Comment:
This should probably be hard coded to arrows, since jump is also bound to 'z' by default.

Reply:
I agree, main menu controls should be their own input. However, the current implementation of the input mapper does not permit binding one key to multiple inputs. I'll open an issue for updating this.